index.html

```diff
@@ -89,44 +143,166 @@
 <script>
   var data = [];
   var options = {
-    yearRange: 'all'
+    categoryRange: 'all'
   };
+  
+
+  var svg, x, y;
 
-  var svg;
+  var margin = { top: 15, right: 15, bottom: 30, left: 45 } ; //spacing b/w svg and g
+
+  var width = 840 - margin.right - margin.left;
+  var height = 500 - margin.top - margin.bottom;
 
   // FETCH DATA
 
+  d3.json('school_data.json', function(error, json) {
+    if (error) { throw json;} //if error is defined, then don't do anything
+
+
+    
+
+    data = json;
+
+    setup(); //must occur after fetching the data
+    update();
+  });
+
   function setup() {
     // SVG
+   
 
     svg = d3.select('#chart')
-      .append('svg');
+      .append('svg')
+      .attr('width', width + margin.right + margin.left)
+      .attr('height', height + margin.top + margin.bottom)
+      .append('g')
+      .attr('transform', 'translate(' + margin.left + ',' + margin.top + ')');
 
+
+ 
     // SCALES
 
+    var padding = 20;
```

Comment:
Padding's an alternative to the margin pattern, we probably don't need to use both, and that's likely making the math more complicated.